site/viamguides/src/air-quality/air-quality.md

```diff
@@ -0,0 +1,340 @@
+    > If you're curious about what these values mean, refer to the U.S. Environmental Protection Agency's [basics on particulate matter (PM)](https://www.epa.gov/pm-pollution/particulate-matter-pm-basics). For measuring the [indoor air quality](https://www.epa.gov/indoor-air-quality-iaq/indoor-pollutants-and-sources) in your home, we'll be looking at PM2.5, a very small particle size.
+
+### Test the smart plug
+
+1. Next, test that your smart plug is working properly. Make sure the fan is still plugged in to the powered smart plug. Also make sure the fan knob is set to on. 
+1. In [the Viam app](https://app.viam.com/robots) under the **CONTROL** tab, find the `generic-1` card, and expand the DOCOMMAND accordion. The module we added has a few functions to control the smart plug. Under the **Input** section, add the following JSON object (if your fan is still running), and then click **Execute** to toggle off the smart plug.
+    ```json
+    {
+      "toggle_off": []
+    }
+    ```
+    ![do command](assets/doCommand.png)
+1. Now update `"toggle_off"` to `"toggle_on"` in the JSON object, and **Execute** once again.
+1. From [the module listing in the registry](https://app.viam.com/module/joyce/kasasmartplug), you may be able to refer to additional documentation in [the module's README](https://github.com/loopDelicious/viam-kasa-switch) to learn about other functions you can use to test your hardware.
+
+<!-- ------------------------ -->
+## Program your air monitoring device
+Duration: 6
+
+At this point, you have configured and tested your machine and peripherals, but nothing is happening automatically. In the next section, program your air monitoring device to be a little smarter. Make your machine periodically get readings, turn on the air filter if the values are high, and turn off again when the values are low.
+
+### Create an automation script
+
+1. To configure the machine to automatically run a command to execute a script, use a [Viam process](https://docs.viam.com/configure/processes/). Create a new file called `process.py`.
+    ```bash
+    $ touch process.py
+    ```
+1. Copy and paste [this sample code](https://github.com/loopDelicious/viam-pm25-process/blob/main/process.py) into the new file `process.py`. This code will allow your Raspberry Pi to connect to both our sensor and plug and execute our logic.
+1. Now it's time to move your control code to your Raspberry Pi device. [SSH into your Raspberry Pi](https://docs.viam.com/installation/prepare/rpi-setup/#connect-with-ssh) if you're not already SSH'd.
+1. From the SSH prompt on your Raspberry Pi, create a directory called `process` to hold your code. And then, change into that directory.
+    ```bash
+    $ mkdir process
+    $ cd process
+    ```
+1. Install the Python package manager.
+    ```bash
+    $ sudo apt install python3-pip
+    ```
+1. Install the Viam Python SDK **into that new directory**.
+    ```bash
+    $ pip3 install --target=process viam-sdk
```

Comment:
If the reader changes directories to be in the `process` directory, passing the `--target` flag will create a new subdirectory called `process`, in which pip will install the `viam-sdk` with its dependencies, so we could probably shorten this up to remove the first step of making the `process` directory and rely on pip to do it:

```markdown
1. Install the Python package manager.
    ```bash
    $ sudo apt install -y python3-pip
    ```
1. Install the Viam Python SDK into a new directory called `process`.
    ```bash
    $ pip3 install --target=process viam-sdk
    ```
```

As an aside, I added the `-y` as part of the `apt install` command to automatically approve the install rather than requiring the reader to hit `y/n` as part of the interactive command.